tools/clang/tools/dxildll/dxcvalidator.cpp

```diff
@@ -0,0 +1,105 @@
+///////////////////////////////////////////////////////////////////////////////
+//                                                                           //
+// dxcvalidatorp.cpp                                                         //
+// Copyright (C) Microsoft Corporation. All rights reserved.                 //
+// This file is distributed under the University of Illinois Open Source     //
+// License. See LICENSE.TXT for details.                                     //
+//                                                                           //
+// Implements the DirectX Validator object, including hashing support.       //
+//                                                                           //
+///////////////////////////////////////////////////////////////////////////////
+
+#include "dxcvalidator.h"
+
+#include "dxc/Support/WinIncludes.h"
+
+#include "dxc/Support/Global.h"
+#include "dxc/Support/microcom.h"
+#include "dxc/dxcapi.h"
+
+using namespace llvm;
+using namespace hlsl;
+
+class DxcValidator : public IDxcValidator2, public IDxcVersionInfo {
```

Comment:
I'm trying to think of a reason why we might want to enable querying the git commit info from this and coming up with nothing, but if you can think of something, maybe we should inherit from `IDxcVersionInfo2`

Reply:
Unable to find a reason as well.

Reply:
The DXIL major/minor version info is not sufficient to identify different validator versions within the same release cycle.  Since this change introduces a build of `dxil.dll` into every build of the public DXC project, it seems more important than before to retain the additional version information going forward.

What do you think?

Reply:
> it seems more important than before to retain the additional version information going forward.

Totally agree with this.  I hope the answer isn't to use a commit hash though.  Ideally we'd get some kind minor-minor version number we can increment as appropriate? Maybe this is something we'll need to consider independently of this though

Reply:
We already have version number in dxc --version like
`dxcompiler.dll: 1.8 - 1.8.0.4710 (test_gcc_bld, 913ae2ce8); dxil.dll: 1.8(1.8.0.4710)`.